scripts/cli_options.py

```diff
@@ -1,13 +1,72 @@
 #! /usr/bin/env python
 from __future__ import absolute_import, print_function
 import random
+from optparse import OptionParser, OptionValueError
+
+import jmclient.support
 
 """This exists as a separate module for two reasons:
-to reduce clutter in main scripts, and (TODO) refactor out
+to reduce clutter in main scripts, and refactor out
 options which are common to more than one script in a base class.
 """
 
-from optparse import OptionParser
+order_choose_algorithms = (
+    'random_under_max_order_choose', 'cheapest_order_choose',
+    'weighted_order_choose')
+
+
+def add_common_options(parser):
+    parser.add_option(
+        '-f',
+        '--txfee',
+        action='store',
+        type='int',
+        dest='txfee',
+        default=-1,
+        help='number of satoshis per participant to use as the initial estimate '
+        'for the total transaction fee, default=dynamically estimated, note that this is adjusted '
+        'based on the estimated fee calculated after tx construction, based on '
+        'policy set in joinmarket.cfg.')
+    parser.add_option('--fast',
+                      action='store_true',
+                      dest='fastsync',
+                      default=False,
+                      help=('choose to do fast wallet sync, only for Core and '
+                            'only for previously synced wallet'))
+    parser.add_option(
+        '-x',
+        '--maxcjfee',
+        type='float',
+        dest='maxcjfee',
+        nargs=2,
+        default=(0.01, 10000),
+        help="maximum coinjoin fee and bitcoin value to pay to a single"
+             "market maker. Both values need to be exceeded, so if the fee is "
+             "30% but only 500satoshi is paid the tx will go ahead. "
+             "default=0.01, 10000 (1%, 10000satoshi)")
+    parser.add_option(
+        '--order-choose-algorithm',
+        action='callback',
+        type=str,
+        default=jmclient.support.random_under_max_order_choose,
```

Comment:
As discussed, I think we can't have this as default, at least to start with. I think we should keep weighted_order_choose as the default for now. The problem is not so much with it being default, as with it being default and the maxcjfee also being default (as it would be for those people who don't add any options).